### PR TITLE
refactor: comments

### PR DESCRIPTION
**Pull request by DNR800**
Looking to apply the zustand state management to comments handling which means we can remove the Provider/Context code

### CODE REVIEW

File: src/shared/stores/data-value-store.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    },
+    getHighlightedField: () => get().highlightedField,
+    isFieldHighlighted: ({ dataElementId, categoryOptionComboId }) =>
+        get().highlightedField?.de?.id === dataElementId &&
+        get().highlightedField?.coc?.id === categoryOptionComboId,
+    setHighlightedField: (value) => {
+        set({ ...get(), highlightedField: value ?? null })
+    },
+    getHighlightedFieldData: () => {
+        const highlightedField = get().getHighlightedField()
```
Comment (DNR800) [inline src/shared/stores/data-value-store.js:32]: Having a centralised store means that you can have more complex selectors - I think there are a couple of places where we "munge" the highlighted item data and data value so potentially you could do this in the selector

Comment (Birkbjo) [reply src/shared/stores/data-value-store.js:32]: I think we should keep them separate, and let the `dataValue`-store be an exact reference to the same object as RQ-return data.

File: src/shared/stores/data-value-store.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    },
+    hasComment: ({ dataElementId, categoryOptionComboId }) => {
+        const dataValue = get().getDataValue({
+            dataElementId,
+            categoryOptionComboId,
+        })
+
+        return !!dataValue?.comment
+    },
+    setComment: ({ dataElementId, categoryOptionComboId, comment }) => {
```
Comment (Birkbjo) [inline src/shared/stores/data-value-store.js:79]: I don't think we should update this store - then we don't share the object-reference anymore.
Also this is done in a mutation, and optimistically updated - so you should be able to just get this from the cache - without having to set it yourself.

File: src/shared/stores/data-value-store.js (hunk 1 above)
Comment (DNR800) [reply src/shared/stores/data-value-store.js:32]: Ok I'll drop the commits for HighlightedField

File: src/shared/stores/data-value-store.js (hunk 2 above)
Comment (DNR800) [reply src/shared/stores/data-value-store.js]: Have remove the use of `setComments`, the old functionality for this (in the provider that has been deleted on this PR) where originally updating a lookup object. checking on the at the dataValue object directly means we don't have to keep anything else up to date.

**Review by Birkbjo — APPROVED** [review]
So much cleaner! LGTM!

**Pull request merged**